certkms/certkms.go

```diff
@@ -0,0 +1,100 @@
+package certkms
+
+import (
+	"context"
+	"crypto"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/psanford/kmssigner"
+	"github.com/slackhq/nebula/cert"
+)
+
+type NebulaSigner interface {
+	// Provides Public() and Sign(...) methods
+	crypto.Signer
+
+	CertSignerLambda() cert.SignerLambda
+}
+
+type nebulaSigner struct {
+	crypto.Signer
+
+	signerLambda cert.SignerLambda
+}
+
+type AWSConfig struct {
+	AssumeRole string
+	Profile    string
+	Region     string
+}
+
+// BasicSigner builds a NebulaSigner from a basic AWS config.
+// More complicated uses can use Signer
+func BasicSigner(config AWSConfig, arn string) (NebulaSigner, error) {
```

Comment:
There is a Go saying, "accept interfaces, return structs"

```suggestion
func BasicSigner(config AWSConfig, arn string) (nebulaSigner, error) {
```